pkg/api/v1/types_swagger_doc.go

```diff
@@ -0,0 +1,1164 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+// This file contains a collection of methods that can be used from go-resful to
+// generate Swagger API documentation for its models. Please read this PR for more
+// information on the implementation: https://github.com/emicklei/go-restful/pull/215
+//
+// TODOs are ignored from the parser. (e.g. TODO(andronat):... || TODO:...) iff
+// are on one line! For multiple line or blocks that you want to ignore use ---.
+// Any context after a --- is ignored.
+//
+// Those methods can be generated by using hack/update-generated-swagger-docs.sh
+
+// AUTO-GENERATED FUNCTIONS START HERE
+func (APIVersion) SwaggerDoc() map[string]string {
+	return map[string]string{
```

Comment:
I would still prefer global variables. It shouldn't be difficult. Instead of generating
```
func (T) SwaggerDoc() map[string]string {
    return (map)
}
```
You do
```
var _T_SwaggerDoc_Map = (map)
func (T) SwaggerDoc() map[string]string { return _T_SwaggerDoc_Map }
```
The map will be initialized at package-initialization and just reused.